server/src/main/java/com/hedera/block/server/persistence/storage/PersistenceStorageConfig.java

```diff
@@ -16,56 +16,154 @@
 
 package com.hedera.block.server.persistence.storage;
 
-import static com.hedera.block.server.Constants.BLOCK_NODE_ROOT_DIRECTORY_SEMANTIC_NAME;
-import static java.lang.System.Logger.Level.ERROR;
+import static com.hedera.block.server.Constants.BLOCK_NODE_ARCHIVE_ROOT_DIRECTORY_SEMANTIC_NAME;
+import static com.hedera.block.server.Constants.BLOCK_NODE_LIVE_ROOT_DIRECTORY_SEMANTIC_NAME;
 import static java.lang.System.Logger.Level.INFO;
 
-import com.hedera.block.common.utils.FileUtilities;
+import com.hedera.block.common.utils.StringUtilities;
 import com.swirlds.config.api.ConfigData;
 import com.swirlds.config.api.ConfigProperty;
+import edu.umd.cs.findbugs.annotations.NonNull;
 import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
 import java.nio.file.Path;
-import java.nio.file.Paths;
+import java.util.Objects;
 
 /**
- * Use this configuration across the persistent storage package
+ * Use this configuration across the persistence storage package.
  *
- * @param rootPath provides the root path for saving block data, if you want to override it need to
- *     set it as persistence.storage.rootPath
+ * @param liveRootPath provides the root path for saving blocks live
+ * @param archiveRootPath provides the root path for archived blocks
  * @param type use a predefined type string to replace the persistence component implementation.
- *     Non-PRODUCTION values should only be used for troubleshooting and development purposes.
+ * Non-PRODUCTION values should only be used for troubleshooting and development purposes.
  */
 @ConfigData("persistence.storage")
 public record PersistenceStorageConfig(
-        @ConfigProperty(defaultValue = "") String rootPath, @ConfigProperty(defaultValue = "PRODUCTION") String type) {
+        @ConfigProperty(defaultValue = "") String liveRootPath,
+        @ConfigProperty(defaultValue = "") String archiveRootPath,
+        @ConfigProperty(defaultValue = "BLOCK_AS_LOCAL_FILE") StorageType type) {
     private static final System.Logger LOGGER = System.getLogger(PersistenceStorageConfig.class.getName());
+    private static final String LIVE_ROOT_PATH =
+            Path.of("hashgraph/blocknode/data/live/").toAbsolutePath().toString();
+    private static final String ARCHIVE_ROOT_PATH =
+            Path.of("hashgraph/blocknode/data/archive/").toAbsolutePath().toString();
```

Comment:
Thanks for noticing this, here my thoughts on this config:

Yes, this is something I wanted to go back to eventually, there is a chore there tagged anyway. This indeed looks off, for sure. Initially, I did not want to escape the already existing logic which did some normalization and had some requirements for the configured path. I did some small refactoring to do the normalization in a couple of methods so that I do not copy-paste now since we have two distinct roots. This whole thing is currently kind of a hack to be able to resolve the blank default more than anything else and I agree it should change.

As far as the `private static final String` fields, I did that because I need to use them in a couple of places and wanted them to be manageable from a single place in case of future needs of modification.

What I wanted to test was if I pass a given `String` as default and/or through other means (i.e. override the default) and have the config field of type `Path` would the config dependency support that out of the box. Thankfully it does, which is awesome! So in theory, we can migrate fairly easy to `Path` from `String` currently. Also, I think we can proactively resolve these paths since this operation will be done only once in the lifetime of the app, once initialized, the config fields will not change. `Path` is also thread safe and immutable by specification so it can be reused without problem.

As far as the constructor, it is currently used solely for normalization of the paths, I do see some preconditions here that makes sense, for example the provided path should in my opinion be absolute, if not it is not deterministic. The good thing about this config is that it is in the initialization phase/lifecycle of the project so this is the best time to blow an exception and prevent the app from starting (fail early) and to fix the problem. With a slight modification already planned for the very near future #371 I can simply add the point to simplify the logic in the constructor to only be a precondition check `if !(live/archive).isAbsolute() then throw`

So, here my proposal (in order to not stall this branch anymore, the refactoring for this will not be that small, there are quite the few usages):
   1. I will extend #371 to also address migration from `String` to `Path` as config types
   1. I will extend #371 to only have a precondition check `if !(live/archive).isAbsolute() then throw` in the constructor, fail early if not able to pass, and to call `Path#normalize`
   1. We will consider a misconfiguration by the operator for any values supplied externally that are not able to pass the precondition check

Reply:
For (2) I would _not_ check for absolute path (elsewhere I've suggested having a base path and relative sub-paths).
I am not certain we actually need a precondition (other than, _perhaps_ an existence check).
A note on determinism: We should _not_ try too hard to be deterministic.  Both deployment and operation of the Block Node are permitted, in most areas, to not be deterministic, so trying to make everything completely deterministic and repeatable is an unnecessary constraint that leads to added complexity.

Reply:
I have extended #371 and it is planned in the near future.